shared/common-adapters/list2.native.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import React, {PureComponent} from 'react'
+import {FlatList, View} from 'react-native'
+import {globalStyles, collapseStyles, styleSheetCreate} from '../styles'
+import {memoize} from '../util/memoize'
+
+import type {Props} from './list2'
+
+class List2 extends PureComponent<Props<any>, void> {
+  static defaultProps = {
+    keyboardShouldPersistTaps: 'handled',
+  }
+  _itemRender = ({item, index}) => {
+    return this.props.renderItem(index, item)
+  }
+
+  _memoizedOffsetGetter = memoize((data, index) => {
+    if (index === 0) {
+      return 0
+    }
+    if (!index) {
+      return 0
+    }
+    // $FlowIssue can't figure out type for data.
+    return this._memoizedOffsetGetter(index - 1) + this.props.itemHeight(index, data[index])
+  })
+
+  _getItemLayout = (data, index) => {
```

Comment:
i don't think we likely even want to make a value for this, it can be a prop passed in. its fine for the fixed height version but we likely should skip the variable case

Reply:
Sorry I'm not sure I understand what you mean. There's a `itemHeight: number | (index: number, item: Item) => number` passed in through `props`. This function just converts the format into what the native list wants.

Reply:
this function is optional to the underlying flat list. we don't/can't generalize this in the generic case so lets not implement it by default. we can implement it if its a fixed size since we can implement it correctly. we can chat about htis